src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Unix.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        private static string s_osDescription = null;
+        private static readonly bool s_is64BitProcess = IntPtr.Size == 8;
+        private static object s_osLock = new object();
+        private static object s_processLock = new object();
+        private static Architecture? s_osArch = null;
+        private static Architecture? s_processArch = null;
+
+        public static string OSDescription
+        {
+            get
+            {
+                if (null == s_osDescription)
```

Comment:
Locks are used where nullable enums are assigned, as assinging to nullable enums is not thread safe. it requires setting hasValue and value, it could happen that a thread sets value and before hasValue is set to true, another thread might read value, which will throw if hasValue is false. Assigning to a string is thread safe, hence no locks are used.